packages/amplify-category-geo/src/service-stacks/baseStack.ts

```diff
@@ -42,9 +42,13 @@ export class BaseStack extends cdk.Stack {
    * converts to cfn
    */
   toCloudFormation = (): $TSAny => {
-    cdkV1PrepareAppShim(this);
-    const cfn = this._toCloudFormation();
-    return cfn;
+    const root = this.node.root as cdk.Stage;
+    const assembly = root.synth();
+    if (!this.nestedStackParent) {
+      return assembly.getStackArtifact(this.artifactId).template;
+    }
+    // if this is a nested stack ( i.e. it has a parent), then just read the template as a string
+    return JSON.parse(fs.readFileSync(path.join(assembly.directory, this.templateFile)).toString('utf-8'));
```

Comment:
Can you add comments about origin of this. 
Something like https://github.com/aws-amplify/amplify-category-api/blob/ede462ea691872c8a8f7d28cefffa6da2fe57c5a/packages/amplify-category-api/src/provider-utils/awscloudformation/base-api-stack.ts#L504-L507